client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/DataFrameAnalyticsConfig.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.ml.dataframe;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentParserUtils;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ObjectParser.ValueType.OBJECT_ARRAY_BOOLEAN_OR_STRING;
+import static org.elasticsearch.common.xcontent.ObjectParser.ValueType.VALUE;
+
+public class DataFrameAnalyticsConfig implements ToXContentObject {
+
+    public static DataFrameAnalyticsConfig fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null).build();
+    }
+
+    private static final String NAME = "data_frame_analytics_config";
+
+    private static final ParseField ID = new ParseField("id");
+    private static final ParseField SOURCE = new ParseField("source");
+    private static final ParseField DEST = new ParseField("dest");
+    private static final ParseField ANALYSIS = new ParseField("analysis");
+    private static final ParseField CONFIG_TYPE = new ParseField("config_type");
+    private static final ParseField ANALYZED_FIELDS = new ParseField("analyzed_fields");
+    private static final ParseField MODEL_MEMORY_LIMIT = new ParseField("model_memory_limit");
+
+    private static ObjectParser<Builder, Void> PARSER = new ObjectParser<>(NAME, true, Builder::new);
+
+    static {
+        PARSER.declareString((c, s) -> {}, CONFIG_TYPE);
```

Comment:
nit: `PARSER` is a lenient parser it is constructed with `ignoreUnknownFields =  true` so it would ignore the `config_type` field even if it wasn't declared here

Reply:
Done.